app/controllers/api/v1/historical_sol_prices_controller.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
```

Comment:
All of these should have gone to api>sol_prices_controller, and modify existing code only where necessary.
Now we have two controllers that do almost the same. Please fix that.